packages/api/src/utils/uploader/r2-uploader.js

```diff
@@ -1,27 +1,28 @@
 import pRetry from 'p-retry'
+import { sha256 } from 'multiformats/hashes/sha2'
 import { toString as uint8ArrayToString } from 'uint8arrays/to-string'
+import { createCarIndex } from '../car.js'
 
 /**
  * @typedef {import('../../bindings').Uploader} Uploader
  * @implements {Uploader}
  */
 export class R2Uploader {
   /**
-   * @param {R2Bucket} bucket
-   * @param {string} publicUrl
+   * @param {object} opts
+   * @param {R2Bucket} opts.carpark
+   * @param {R2Bucket} opts.dudewhere
+   * @param {R2Bucket} opts.satnav
+   * @param {string} opts.publicUrl
```

Comment:
When it's not optional options, I like to call it config...

```suggestion
   * @param {object} config
   * @param {R2Bucket} config.carpark
   * @param {R2Bucket} config.dudewhere
   * @param {R2Bucket} config.satnav
   * @param {string} config.publicUrl
```